src/NHibernate/Hql/Ast/ANTLR/Tree/FromClause.cs

```diff
@@ -72,15 +72,27 @@ public FromClause ParentFromClause
 		{
 			get { return _parentFromClause; }
 		}
-
+		
+		//6.0 TODO: Replace with Typed version below
 		public IList<IASTNode> GetExplicitFromElements()
 		{
-			return ASTUtil.CollectChildren(this, ExplicitFromPredicate);
+			return ASTUtil.CollectChildren<IASTNode>(this, ExplicitFromPredicate);
 		}
-
+		
+		internal IList<FromElement> GetExplicitFromElementsTyped()
```

Comment:
It feels like the return type shall be `IReadOnlyList<>`.

Reply:
Why? That's not something we care about. Maybe user want to shuffle or slightly modify results in some way before further processing. I see no point in such restriction.

Reply:
I do not think we should restrict a method return type to be readonly when altering the returned value does not have any impact on the state of the object holding the method.